spring-session-core/src/main/java/org/springframework/session/MapSession.java

```diff
@@ -229,7 +231,14 @@ public int hashCode() {
 	}
 
 	private static String generateId() {
-		return UUID.randomUUID().toString();
+		return sessionIdGenerator.generateId();
+	}
+
+	private static final SessionIdGenerator sessionIdGenerator;
```

Comment:
`static` is not good idea

Reply:
Then what's your suggestion?

Reply:
At present, this PR can solve the problem easily. 
I think `static` is structural problem (not this PR), I prefer PR #1547, using spring DI is better than SPI.

Reply:
That huge changeset is risky, this one is simple and reliable.